Win/Mineways.cpp

```diff
@@ -2460,6 +2501,135 @@ static void copyOverExportPrintData( ExportFileData *pEFD )
+
+    gOptions.exportFlags |=
+        (gpEFD->chkHollow[gpEFD->fileType] ? EXPT_HOLLOW_BOTTOM : 0x0) |
+        ((gpEFD->chkHollow[gpEFD->fileType] && gpEFD->chkSuperHollow[gpEFD->fileType]) ? EXPT_HOLLOW_BOTTOM|EXPT_SUPER_HOLLOW_BOTTOM : 0x0);
+
+    gOptions.exportFlags |= EXPT_OUTPUT_OBJ_GROUPS; // export groups
+    gOptions.exportFlags |= EXPT_OUTPUT_OBJ_MATERIAL_PER_TYPE; // the norm, instead of single material
+    gOptions.exportFlags |= EXPT_OUTPUT_OBJ_FULL_MATERIAL; // Full material (output the extra values)
+    gOptions.exportFlags |= EXPT_OUTPUT_TEXTURE_IMAGES; // Export block full texture
+    gOptions.exportFlags |= EXPT_OUTPUT_OBJ_REL_COORDINATES; // OBj relative coordinates
+    gOptions.exportFlags |= EXPT_BIOME; // Use biome for export. Currently only the biome at the center of the zone is supported
+
+    // Generate files
+    FileList outputFileList;
+    outputFileList.count = 0;
+    if ( on ) {
+
+        draw();
+        InvalidateRect(hWnd, NULL, FALSE);
+        UpdateWindow(hWnd);
+        gpEFD->radioScaleToHeight = 1;
+        gpEFD->radioScaleByCost = 0;
+
+        int errCode = SaveVolume(objFileName, gpEFD->fileType, &gOptions, gWorld, gCurrentDirectory,
+            gpEFD->minxVal, gpEFD->minyVal, gpEFD->minzVal, gpEFD->maxxVal, gpEFD->maxyVal, gpEFD->maxzVal,
+            updateProgress, terrainFileName, &outputFileList, (int)gMajorVersion, (int)gMinorVersion);
+
+        // Build the .zip: Note that the first file saved in ObjManip.c is the one used as the zip file's name
+        wchar_t wcZip[MAX_PATH];
+        LPTSTR tempdir = new TCHAR[MAX_PATH];
+
+        // Write zip in temp directory
+        GetTempPath(MAX_PATH, tempdir);
+        swprintf_s(wcZip, MAX_PATH, L"%s\\%s.zip", tempdir, outputFileList.name[0]);
+        DeleteFile(wcZip);
+
+        HZIP hz = CreateZip(wcZip,0,ZIP_FILENAME);
+        for ( int i = 0; i < outputFileList.count; i++ )
+        {
+            const wchar_t *nameOnly = removePath( outputFileList.name[i] ) ;
+
+            if (*updateProgress)
+            { (*updateProgress)(0.10f + 0.10f*(float)i/(float)outputFileList.count);}
+
+            ZipAdd(hz,nameOnly, outputFileList.name[i], 0, ZIP_FILENAME);
+
+            // delete model files if not needed
+            if ( !gpEFD->chkCreateModelFiles[gpEFD->fileType] )
+            {
+                DeleteFile(outputFileList.name[i]);
+            }
+        }
+        CloseZip(hz);
+
+        if (*updateProgress)
+            (*updateProgress)(1.0f);
+
+        if ( errCode != MW_NO_ERROR )
+        {
+            PopupErrorDialogs( errCode );
+        }
+
+        // Set filepath to skfb data
+        std::wstring file(wcZip);
+        std::string filepath(file.begin(), file.end());
+        skfbPData->skfbFilePath = filepath;
+		setPublishSkfbData(skfbPData);
+
+        uploadToSketchfab(hInst, hWnd);
+    }
+
+    return retCode;
+}
```

Comment:
That's true, it would be better to split all of this in separate functions. The "problem" with this is that i would also imply some refactoring in existing code (since the process is "almost" the same), but I  don't really now how good it would be to have such changes in existing code in our diff. Maybe I am wrong but limiting the diff to our own changes, and avoiding changes in existing code (without changing the behavior), looks better